Explorer/Assets/DCL/SDKComponents/Utils/FindAvatarUtils.cs

```diff
@@ -0,0 +1,73 @@
+using Arch.Core;
+using DCL.AvatarRendering.AvatarShape.Components;
+using DCL.AvatarRendering.AvatarShape.UnityInterface;
+using DCL.Multiplayer.Connections.Typing;
+using DCL.Multiplayer.Profiles.Tables;
+using System.Diagnostics;
+using UnityEngine;
+using Utility.Multithreading;
+
+namespace DCL.SDKComponents.Utils
+{
+    public static class FindAvatarUtils
+    {
+        private static readonly QueryDescription AVATAR_BASE_AND_SHAPE_QUERY = new QueryDescription().WithAll<AvatarBase, AvatarShapeComponent>();
+        private static readonly QueryDescription AVATAR_BASE_QUERY = new QueryDescription().WithAll<AvatarBase>();
+        public static IReadOnlyEntityParticipantTable? EntityParticipantTable;
+
+        [Conditional("UNITY_EDITOR")]
+        [Conditional("DEBUG")]
+        private static void AssertMainThread() =>
+            MultithreadingUtility.AssertMainThread(nameof(FindAvatarUtils), true);
+
+        public static LightResult<AvatarBase> AvatarWithID(World globalWorld, string id)
+        {
+            AssertMainThread();
+
+            // Try to find the avatar using the EntityParticipantTable
+            if (EntityParticipantTable != null)
+            {
+                if (EntityParticipantTable.TryGet(id, out IReadOnlyEntityParticipantTable.Entry entry)
+                    && globalWorld.TryGet(entry.Entity, out AvatarBase? avatarBase))
+                    return new LightResult<AvatarBase>(avatarBase!);
+
+                return LightResult<AvatarBase>.FAILURE;
+            }
+
+            // Fall back to the less performant ECS query approach if the entity participant table is not available
```

Comment:
Is it still a valid pass if `EntityParticipantTable` is made always available? Is it possible to have `AvatarShapeComponent` outside `EntotyParticipantTable`, e.g. spawned by the scene?

Reply:
Uhmm you are right that for scene avatar shapes I should take into account those missing participants in the table, so I'll update it

Reply:
Tackled at https://github.com/decentraland/unity-explorer/pull/3430/commits/8861ff81ea16779ee379fb547fd0353f842945fc